addons/website/models/ir_ui_view.py

```diff
@@ -298,15 +298,21 @@ def filter_duplicate(self):
         if not current_website_id:
             return self.filtered(lambda view: not view.website_id)
 
+        views_keys = set()
+        for view_check in self:
+            if view_check.website_id and view_check.website_id.id == current_website_id:
+                views_keys.add(view_check.key)
+
         for view in self:
             # specific view: add it if it's for the current website and ignore
             # it if it's for another website
             if view.website_id and view.website_id.id == current_website_id:
                 most_specific_views |= view
             # generic view: add it only if, for the current website, there is no
-            # specific view for this view (based on the same `key` attribute)
-            elif not view.website_id and not any(view.key == view2.key and view2.website_id and view2.website_id.id == current_website_id for view2 in self):
-                most_specific_views |= view
+            # specific view for this view (based on the same key attribute)
```

Comment:
Very small detail but... don't change that comment. History change for no reason... and the comment was actually better before, no? :shrug: